File: vue.jsonforms/src/entities/owners-and-controllers/IndividualControllerDetails.ts
import AddressInfo from "@/pages/models/owners-and-controllers/AddressInfo";
import { ContactNumber } from "../ContactNumber";
import {IShareholder} from "@/entities/firm-details/IShareholder";

export class IndividualControllerDetails implements IShareholder {
  public title: string | undefined;
  public forename: string | undefined;
  public surname: string | undefined;
  public previousFullName: string | undefined;
  public reasonForChangeName: string | undefined;
  public dateOfNameChange: number = 0;
  public commonlyUsedName: string | undefined;
  public contactNumber: ContactNumber = new ContactNumber();
  public emailAddress: string | undefined;
  public homeAddress: string | undefined;
  public isHomeAddressChanged: boolean = false;
  public homeAddressResidenceDate: number | undefined;
  public previousAddresses: AddressInfo[] = [];
  public dateOfBirth: number = 0;
  public countryOfBirth: string | undefined;
  public nationalities: string[] = [];
  public previousNationalities: string[] = [];
  public nationalInsuranceNumber: string | undefined;
  public passportNumber: string | undefined;
  public percentageOfCapital: number | null = null;
  public percentageOfVotingRights: number | null = null;
  public hasBeenSubjectToAnyMaterialComplaints?: boolean = undefined;
  public additionalInformation: string | undefined;
  public supportingDocumentsUrls: string[] = [];
  public originalHomeAddress: string | undefined;
}

File: CSharp.Functions/Common/Entities/FirmDetails/OwnersAndControllers/IndividualControllerDetails.cs
namespace Common.Entities
{
    public class IndividualControllerDetails
    {
        public string? Title { get; set; }
        public string? Forename { get; set; }
        public string? Surname { get; set; }
        public string? PreviousFullName { get; set; }
        public string? ReasonForChangeName { get; set; }
        public long DateOfNameChange { get; set; }
        public string? CommonlyUsedName { get; set; }
        public ContactNumber? ContactNumber { get; set; }
        public string? EmailAddress { get; set; }
        public string? HomeAddress { get; set; }
        public bool IsHomeAddressChanged { get; set; }
        public long? HomeAddressResidenceDate { get; set; }
        public List<AddressInfo> PreviousAddresses { get; set; } = new();
        public long DateOfBirth { get; set; }
        public string? CountryOfBirth { get; set; }
        public List<string> Nationalities { get; set; } = new();
        public List<string> PreviousNationalities { get; set; } = new();
        public string? NationalInsuranceNumber { get; set; }
        public string? PassportNumber { get; set; }
        public double? PercentageOfCapital { get; set; }
        public double? PercentageOfVotingRights { get; set; }
        public bool HasBeenSubjectToAnyMaterialComplaints { get; set; }
        public string? AdditionalInformation { get; set; }
        public List<string> SupportingDocumentsUrls { get; set; } = new();
    }
}